### PR TITLE
Prolong maximum chat message length to 1000 characters

### LINKED ISSUE #778: Increase Chat character limit
I've run into the character limit of 300 characters per chat message fairly frequently. It allows for maybe two to three sentences, which is simply too short to answer any question in depth. I'd consider a limit of at least 1000 characters the bare minimum.

Can we increase it? Can we at least exclude admins from it? Or could we even allow the course admins to set the limit for each course individually?

### PR DESCRIPTION
**Pull request by JosefSchoenberger**
### Motivation and Context
We want longer chat messages! I can't answer to the questions of our students if I'm not able to write a long prose with at least 800 characters! :-P
Resolves #778.

### Description
I replaced the chat input box with a multiline text area and extended the maximum message length from 200 to 1000 characters.

### Steps for Testing
1. Log in
2. Find a stream or VOD you can comment on
3. Write a long text message with more then 200 characters. May include multiple lines with Shift+Enter.

### CODE REVIEW

File: web/ts/components/chat-prompt.ts [hunk 1]
```diff
@@ -84,6 +87,10 @@ export function chatPromptContext(streamId: number): AlpineComponent {
             } else if (this.emojis.hasSuggestions()) {
                 this.emojiEl.focus();
             }
+
+            // https://stackoverflow.com/a/21079335
+            this.inputEl.style.height = "0px";
+            this.inputEl.style.height = (this.inputEl.scrollHeight) + "px";
```
Comment (joschahenningsen) [inline web/ts/components/chat-prompt.ts]:
```suggestion
            this.inputEl.style.height = this.inputEl.scrollHeight + "px";
```

**Review by joschahenningsen — CHANGES_REQUESTED** [review]
Looks reasonable, just a quick remark found by eslint:

**Comment by joschahenningsen** [comment]
BTW, there also is server side validation: https://github.com/joschahenningsen/TUM-Live/blob/dev/model/chat.go#L40

**Comment by JosefSchoenberger** [comment]
Alright, thanks – I was already wondering, but the server accepted messages with ~220 characters and did not find any other occurence of "200" related to chat messages, so I assumed there was no server-side check.
Btw., was there a reason for why the server-side limit was higher?

**Comment by joschahenningsen** [comment]
> Btw., was there a reason for why the server-side limit was higher?

I can't think of any reason why we would have done this on purpose. I assume it was just an oversight.

**Review by joschahenningsen — APPROVED** [review]
Thanks!

**Pull request merged**